source/renderer/app/stores/WalletsStore.js

```diff
@@ -308,24 +311,29 @@ export default class WalletsStore extends Store {
     this.setupCurrency();
   }
 
-  @action _getWalletPublicKey = async () => {
+  @action _getPublicKey = async ({
+    spendingPassword: passphrase,
+  }: {
+    spendingPassword: string,
+  }) => {
     if (!this.active || !IS_WALLET_PUBLIC_KEY_SHARING_ENABLED) {
       return;
     }
 
-    // @TODO Once the api is ready, role and index values should be configured properly
+    // @TODO Once the api is ready, index value should be configured properly
     const walletId = this.active.id;
-    const role = '';
-    const index = '';
+    const index = '1852H';
+    const extended = true;
 
     try {
       const walletPublicKey = await this.walletPublicKeyRequest.execute({
         walletId,
-        role,
         index,
+        passphrase,
+        extended,
       }).promise;
       runInAction('update wallet public key', () => {
-        this.activePublicKey = walletPublicKey;
+        this.walletsPublicKeys[walletId] = walletPublicKey;
```

Comment:
@aleksandardjordjeviciohk this needs to be reverted!

Reply:
Reverted